File: 2580-circular-sentence/circular-sentence.cpp
class Solution {
private:
    vector<string> split(string &s, const char &c) {
        int n = s.size(), l = 0, r = 0;
        s.push_back(c);
        vector<string> res;
        do {
            while (s[r] != c) r++;
            string word = s.substr(l, r - l);
            if (!word.empty()) {
                res.push_back(word);
            }
            while (s[r] == c) r++;
            l = r;
        } while (r < n);
        return res;
    }
public:
    bool isCircularSentence(string &s) {
        if (s[0] != s.back()) return false;
        vector<string> arr = split(s, ' ');
        for (int i = 0; i < arr.size() - 1; i++) {
            if (arr[i].back() != arr[i + 1][0]) {
                return false;
            }
        }
        return true;
    }
};

File: 2580-circular-sentence/circular-sentence.py
class Solution:
    def isCircularSentence(self, s: str) -> bool:
        l=s.strip().split(' ')
        n=len(l)
        if n==1:
            return l[0][0]==l[0][-1]
        for i in range(1,n):
            if l[i-1][-1]!=l[i][0] :
                return False
        return l[-1][-1]==l[0][0]
        